### PR TITLE
Fixes service freezes when loading non-module maps

### PR DESCRIPTION
**Pull request by florian-kuebler**
Fixes b/156886143

### CODE REVIEW

File: OrbitCore/LinuxUtils.cpp [hunk 1]
```diff
@@ -149,6 +146,9 @@ void ListModules(pid_t pid,
     if (elf_file->GetBuildId().empty()) continue;
```
Comment (dimitry-) [inline OrbitCore/LinuxUtils.cpp:145]: do we really want to filter out these?

Comment (ronaldfw) [reply OrbitCore/LinuxUtils.cpp:145]: Could you provide more details why you wouldn't want to filter these out?

Comment (dimitry-) [reply OrbitCore/LinuxUtils.cpp:145]: If a file is missing build id it does not necessarily mean we are unable to load symbols. If it has .symtab section available we do not really care if it has a build id. Also we might ending up providing user incomplete information about list of loaded modules. In sampling view module name will be empty if samples end up in such a librarly.

Comment (florian-kuebler) [reply OrbitCore/LinuxUtils.cpp:145]: From the code, I'd assume that they were intended to be filtered out. However, I am also fine with keeping them.

Comment (florian-kuebler) [reply OrbitCore/LinuxUtils.cpp:145]: Okay, so I'll move the `insert_or_assign` above the `if (elf_file->GetBuildId().empty()) continue;`. So the handling of cases where the elf_file has no build_id would be handled as before.

**Comment by dimitry-** [comment]
It fixes a problem.. I believe the more correct way would be to filter out non-executable maps, instead of relying on m_PdbSize being 0.

**Comment by florian-kuebler** [comment]
> It fixes a problem.. I believe the more correct way would be to filter out non-executable maps, instead of relying on m_PdbSize being 0.

Do you have a suggestion of how to detect this nicely?

File: OrbitCore/LinuxUtils.cpp [hunk 2]
```diff
@@ -129,12 +129,8 @@ void ListModules(pid_t pid,
   }
```
Comment (dimitry-) [inline OrbitCore/LinuxUtils.cpp:130]: You will need to parse flags in tokens[1] they have following format "rwxp" where every position can be '-'. the executable flag is in 3rd position.

So the map entry is executable if tokens[1][2] == 'x'

Comment (florian-kuebler) [reply OrbitCore/LinuxUtils.cpp:134]: Thanks. I added a check and discard them already here. I left the below check, just to be sure.

File: OrbitCore/LinuxUtils.cpp [hunk 3]
```diff
@@ -109,6 +109,10 @@ void ListModules(pid_t pid,
     // mapped to a file (might be heap, stack or something else)
     if (tokens.size() != 6 || tokens[4] == "0") continue;
 
+    // tokens[1] is the permissions column. tokens[1][2] is the permission to
+    // be executed. We discard entries that are not executable.
+    if (tokens[1].size() != 4 || tokens[1][2] != 'x') continue;
+
```
Comment (dimitry-) [inline OrbitCore/LinuxUtils.cpp:115]: ok, it appears to be a bit more complicated... if you look into line 121-131, it collects address ranges for maps for every corresponding module. each file usually has 3 entries in /proc/pid/maps the first one is executable, second one is readable and the last one is read-writable. filtering 2 others out will result in some space that is occupied by a service unaccounted.

I think it is better to collect executable flags and after processing everything filtering out files which were not mapped as executables, meaning that all the maps for the file are non-executable.

Comment (florian-kuebler) [reply OrbitCore/LinuxUtils.cpp]: Should be addressed now.

**Pull request merged**